crates/net/discv4/src/config.rs

```diff
@@ -247,6 +253,18 @@ impl Discv4ConfigBuilder {
         self
     }
 
+    /// Adds multiple trusted nodes
+    pub fn add_trusted_nodes(&mut self, nodes: impl IntoIterator<Item = NodeRecord>) -> &mut Self {
+        self.config.trusted_nodes.extend(nodes);
+        self
+    }
+
+    /// Sets flag for using only trusted nodes
+    pub fn set_connect_trusted_nodes_only(&mut self, trusted_only: bool) -> &mut Self {
+        self.config.connect_trusted_nodes_only = trusted_only;
+        self
+    }
```

Comment:
Does this sort of clash with #495 @mattsse? In the sense that if we only want to connect to trusted nodes, we should just connect to them via e.g. `NetworkHandle` and skip discv4 entirely, no?

Reply:
yeh, mentioned here https://github.com/paradigmxyz/reth/pull/569#discussion_r1055352612 trusted nodes should not be integrated with Discovery and kept separately.